synapse/lib/layer.py

```diff
@@ -4093,6 +4094,105 @@ async def iterNodeDataKeys(self, buid):
             prop = self.getAbrvProp(abrv)
             yield prop[0]
 
+    async def iterLayerAddPerms(self):
+
+        # nodes & props
+        for byts, abrv in self.propabrv.slab.scanByFull(db=self.propabrv.name2abrv):
+            form, prop = s_msgpack.un(byts)
+            if form is None:
+                continue
+
+            for _ in self.layrslab.scanByPref(abrv, db=self.byprop):
```

Comment:
Can you just use prefexists() for these?

Reply:
good point, I forgot that existed